.github/workflows/backend.yml

```diff
@@ -103,6 +103,102 @@ jobs:
       - name: Handle artifacts
         uses: ./.github/actions/artifacts
 
+  backend-test-in-control-mode:
+    if: needs.files-changed.outputs.backend == 'true'
+    needs: files-changed
+    name: backend test
```

Comment:
I would update these name attributes so that they're clearly differentiated in the github actions UI. I also question the need to keep the matrix tests for this. At least for now.

Reply:
Ah, good catch. I think I did that in an earlier draft PR like this one and forgot to port it over.

I also agree about the matrix tests (to the extent I even understand what they do 😅). This is currently a very naive copy-and-paste job. Before it becomes real, I'll want to clean up the cruft, as well as seeing if there's a way to abstract out some of the repetition.